src/plugins/intel_gna/src/backend/gna_limitations.cpp

```diff
@@ -15,12 +15,215 @@
 #include "gna/gna_config.hpp"
 #include "gna_graph_tools.hpp"
 #include "gna_lib_ver_selector.hpp"
+#include "ie_ngraph_utils.hpp"
 #include "log/log.hpp"
+#include "ops/util/util.hpp"
 
 namespace ov {
 namespace intel_gna {
 using namespace common;
 namespace limitations {
+namespace {
+std::ostream& operator<<(std::ostream& os, const std::set<ov::element::Type>& t) {
+    for (auto it = t.begin(); it != t.end(); ++it) {
+        if (it != t.begin()) {
+            os << ", " << *it;
+        } else {
+            os << *it;
+        }
+    }
+    return os;
+}
+}  // namespace
+
+const std::set<ov::element::Type> SupportedElementTypes::supported_parameter_types = {ov::element::u8,
+                                                                                      ov::element::i16,
+                                                                                      ov::element::f32};
+
+bool SupportedElementTypes::is_parameter_type_supported(ov::element::Type elem_type, bool is_exception_allowed) {
```

Comment:
Code is fine, but It is good to avoid nested condition if it is not necessary.
```
bool SupportedElementTypes::is_parameter_type_supported(ov::element::Type elem_type, bool is_exception_allowed) {
    if (supported_parameter_types.count(elem_type) > 0) {
        return true;
    }

    if (is_exception_allowed) {
        THROW_GNA_EXCEPTION << "The plugin does not support input precision with " << elem_type.get_type_name()
                            << " format. Supported precisions " << supported_parameter_types << "\n";
    }

    return false;
}
```